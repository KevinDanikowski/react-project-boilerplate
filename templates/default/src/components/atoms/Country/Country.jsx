import React from 'react'
import PropTypes from 'prop-types'

const Country = ({ name }) => {
  return (
    <div className="country">
      <div data-testid="country-name">{name}</div>
      <style jsx>
        {`
          .country {
            border: dashed 1px gray;
            padding: 10px;
            font-weight: 600;
            font-size: 16px;
          }
        `}
      </style>
    </div>
  )
}

Country.propTypes = {
  name: PropTypes.string
}

Country.defaultProps = {
  name: ''
}

export default Country
