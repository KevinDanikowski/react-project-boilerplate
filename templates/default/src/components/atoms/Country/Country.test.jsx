import React from 'react'
import { render } from '@testing-library/react'
import Country from '.'

describe('Country', () => {
  it('should render a Country', () => {
    const { asFragment, getByTestId } = render(<Country name="United States" />)
    const name = getByTestId('country-name')

    expect(name.textContent).toBe('United States')
    expect(asFragment()).toMatchSnapshot()
  })
})
