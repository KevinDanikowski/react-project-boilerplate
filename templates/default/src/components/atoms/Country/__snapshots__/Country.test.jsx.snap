// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country should render a Country 1`] = `
<DocumentFragment>
  <div
    class="jsx-563423332 country"
  >
    <div
      class="jsx-563423332"
      data-testid="country-name"
    >
      United States
    </div>
  </div>
</DocumentFragment>
`;
